src/solvers/refinement/string_refinement.cpp

```diff
@@ -326,85 +453,127 @@ std::size_t expr_cast<std::size_t>(const exprt& val_expr)
 /// same value as the next index that is present in the index set.
 /// We do so by traversing the array backward, remembering the
 /// last value that has been initialized.
-void string_refinementt::concretize_string(const exprt &expr)
+static void concretize_string(
+  const std::function<exprt(const exprt&)> get,
+  std::map<exprt, exprt> &found_length,
+  std::map<exprt, array_exprt> &found_content,
+  const replace_mapt &symbol_resolve,
+  const std::map<exprt, std::set<exprt>> &index_set,
+  const std::size_t max_string_length,
+  messaget::mstreamt &stream,
+  const namespacet &ns,
+  const exprt &expr)
 {
-  if(is_refined_string_type(expr.type()))
+  if(const auto str=expr_cast<string_exprt>(expr))
   {
-    const string_exprt str=to_string_expr(expr);
-    const exprt length=get(str.length());
-    exprt content=str.content();
+    const auto& eom=messaget::eom;
```

Comment:
What's the point of that?

Reply:
`eom`'s used quite often, will store as value.